### DISCUSSION TITLE
有没有什么方法能把ffmpeg从libmpv中独立出来？

### DISCUSSION THREAD

**Discussion by wangwenx190** [zhongfly/mpv-build#66] [General]
现在ffmpeg是静态链接到libmpv中的，导致libmpv的大小非常大。如果能将ffmpeg以动态库的形式独立出来，libmpv可以大大减小，并且独立出来的ffmpeg也可以被其他程序调用，一举多得。

**Comment by zhongfly** [comment]
https://github.com/shinchiro/mpv-winbuild-cmake
你可以在这里提问，我不知道该怎么做。

**Reply by zhongfly** [reply]
另外，我不觉得现在这样有多大…对现在的电脑存储空间来说真是微乎其微

**Reply by wangwenx190** [reply]
我在开发基于libmpv的软件，我还是希望能尽可能小巧一些的，毕竟不是electron之流

**Reply by zhongfly** [reply]
至少可以禁用一些编码器，减小一些体积。别的我就不知道，我对这个的了解可能还没你多。
从开箱即用来看，是不是有可能存在某些ffmpeg构建缺少你需要的功能，结果导致问题…而且有时mpv会提高ffmpeg依赖的最低版本，那用户突然发现不能用了，估计也要花费一些时间来找到原因…个人之见
也许2种libmpv都提供，让用户自己选择也是一种办法
